avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroAnnotationIntrospector.java

```diff
@@ -63,6 +65,15 @@ public PropertyName findNameForDeserialization(Annotated a) {
     }
 
     @Override
+    public Object findDeserializer(Annotated am) {
+        AvroEncode ann = _findAnnotation(am, AvroEncode.class);
+        if (ann != null) {
+            return CustomEncodingDeserializer.class;
```

Comment:
`AnnotationIntrospectorPair` does not support returning `JsonDeserializer` _instances_, so using a `HandlerInstantiator` to instantiate the deserializer with the correct context information.